examples/morpheusvm/tests/transfer.go

```diff
@@ -31,14 +30,13 @@ var _ = registry.Register(TestsRegistry, "Transfer Transaction", func(t ginkgo.F
 	require.NoError(err)
 	toAddress := auth.NewED25519Address(other.PublicKey())
 
-	networkConfig := tn.Configuration().(*workload.NetworkConfiguration)
-	spendingKey := networkConfig.Keys()[0]
-
+	authFactory, err := auth.GetFactory(tn.FundedKey())
```

Comment:
The FundedKey can be access using `NetworkConfiguration.PrivateKeys()[0]`.
- On Network interface, I am wondering if it can make sense to create `FundedAuthFactory()` method instead of `FundedKey()`. This looks less "standard" but it prevents the user from having to generate the authFactory.
- Or another way to simplify his life can be to pass an optional (instead of a required) AuthFactory to GenerateTx(). Then the Network.GenerateTx is responsible to set a default AuthFactory using the first PrivateKey it gets from the NetworkConfiguration.

Reply:
yes we could actually completely replace the `*auth.PrivateKey` type with `AuthFactory` now that it includes an `Address()` function to return the relevant address.

The `PrivateKey` type was introduced to combine both.